source/Firmata/UwpFirmata.cpp

```diff
@@ -45,13 +45,17 @@ namespace {
 
 UwpFirmata::UwpFirmata(
 	void
-	)
+):
+    _blob_started(false),
+    _blob_position(0),
+    _data_buffer(nullptr),
+    _firmata_lock(_firmutex, std::defer_lock),
+    _firmata_stream(nullptr),
+    _input_thread_should_exit(false),
+    _sys_command(0),
+    _sys_position(0)
 {
-	_sysCommand = 0;
-	_dataBuffer = (uint8_t*)malloc( 31 );
-	_sysPosition = 0;
-	_blobStarted = false;
-	_blobPosition = 0;
+	_data_buffer = new uint8_t[31];
```

Comment:
It's good to see everything moving into initializer lists.

You might consider making _data_buffer a std::unique_ptr<uint8_t[]>() which will automatically free the memory in your destructor (and suppress accidental unsafe stuff like copying it).